kii_thing_if.h

```diff
@@ -9,25 +9,41 @@
 extern "C" {
 #endif
 
-/** Enumeration representing error of thing-if ThingSDK. */
-typedef enum kii_thing_if_error_t {
-    /** Functions succeed. There is no error. */
-    KII_THING_IF_ERROR_NO_ERROR = 0,
-    /** Information trying to get from a function is not found. */
-    KII_THING_IF_ERROR_TARGET_NOT_FOUND,
-    /** Thing is not found in Kii Cloud. */
-    KII_THING_IF_ERROR_THING_NOT_FOUND,
+#define KII_THING_IF_TASK_NAME_STATUS_UPDATE "status_update_task"
+
+/** Error reasons of thing-if ThingSDK. */
+typedef enum kii_thing_if_error_reason_t {
     /** kii_thing_if_t instance is not onbarded. Please onboard first. */
-    KII_THING_IF_ERROR_NOT_ONBOARDED,
-    /** A buffer size provided from applications is shorter than a
-     * function can copy the information got from Kii Cloud.
-     */
-    KII_THING_IF_ERROR_LENGTH_EXCEEDED,
+    KII_THING_IF_ERROR_REASON_NOT_ONBOARDED,
     /** thing-if ThingSDK is alreday started. */
-    KII_THING_IF_ERROR_ALREADY_STARTED
-} kii_thing_if_error_t;
+    KII_THING_IF_ERROR_REASON_ALREADY_STARTED,
+    /** HTTP error. */
+    KII_THING_IF_ERROR_REASON_HTTP,
+    /** Socket error. */
+    KII_THING_IF_ERROR_REASON_SOCKET
+} kii_thing_if_error_reason_t;
+
+/** Error information of thing-if ThingSDK. */
+typedef struct kii_thing_if_error_t {
+    /** Error reason. */
+    kii_thing_if_error_reason_t reason;
+
+    /** HTTP status code.
+     *
+     * If ::kii_thing_if_error_t::reason is
+     * ::KII_THING_IF_ERROR_REASON_HTTP, this value is set. Otherwise,
+     * this value is meaningless.
+     */
+    int http_status_code;
 
-#define KII_THING_IF_TASK_NAME_STATUS_UPDATE "status_update_task"
+    /** Error code.
+     *
+     * If ::kii_thing_if_error_t::reason is
+     * ::KII_THING_IF_ERROR_REASON_HTTP, this value is set. Otherwise,
+     * this value is meaningless.
+     */
+    char http_error_code[32];
```

Comment:
`http_error_code` would be confusing.
The error code here is mapped to ERROR_CODE field in response body isn't it?
Since those codes are defined in application layer and HTTP doesn't defines them.